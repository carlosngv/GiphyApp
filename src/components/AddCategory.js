import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState('')

    const _handleOnChange = (e) => {
        setinputValue(e.target.value)
    }

    const _handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2) {
            setCategories((c) => [ inputValue, ...c]);
            setinputValue('');
        }
    }

    return (
        <>
        <form onSubmit={_handleSubmit}>
            <input
                type="text"
                className="input is-rounded"
                value={inputValue}
                onChange={_handleOnChange}
                placeholder='Gif'
            />
        </form>
        </>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}