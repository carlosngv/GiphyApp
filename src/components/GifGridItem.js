import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card animate__animated animate__backInLeft">
            <div className="card-image">
                <figure className="image is-4by3">
                    <img src={url} alt={title} />
                </figure>
            </div>
            <div className="card-content">
                <div className="content">
                    <p>{title}</p>
                </div>
            </div>
        </div>
    )
}


GifGridItem.propTypes =  {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}