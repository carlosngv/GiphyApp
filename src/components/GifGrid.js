import React, { Fragment} from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
         
    return (
        <Fragment>
            <h1 className="category-title animate__animated animate__backInRight">{category}</h1>
            <div className="card-grid">
                    {loading && 'Loading...'} 
                    {
                        
                        images.map((img) => (
                            <GifGridItem
                                key={img.id}
                                {...img}
                            />
                        ))
                    }
            </div>
        </Fragment>
    )
}
