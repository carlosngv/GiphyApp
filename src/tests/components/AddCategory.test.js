import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Tests for AddCategory Component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    
    test('should show AddCategory component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('input should change value correctly', () => {
        const input = wrapper.find('input');
        const value = 'Hello world';
        input.simulate('change', { 
            target: {
                value: value
            } 
        });
    });
    
    test('should appear same value as input within p tag', () => {
        const input = wrapper.find('input');
        const value = 'Hello world';
        input.simulate('change', { 
            target: {
                value: value
            } 
        });
        const pTagContent = wrapper.find('p').text().trim(); 
        expect(pTagContent).toBe(value);
    });

    test('Shouldnt post info on submit', () => {
        wrapper.find('form').simulate('submit', { 
            preventDefault(){} 
        });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and clean input', () => {
        const input = wrapper.find('input');
        const value = 'This is a test';
        input.simulate('change', {
            target: {
                value: value
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });

        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value').trim()).toBe('');

    });

});