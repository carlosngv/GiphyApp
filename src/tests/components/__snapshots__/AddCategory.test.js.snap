// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AddCategory Component should show AddCategory component correctly 1`] = `
<Fragment>
  <p />
  <form
    onSubmit={[Function]}
  >
    <input
      className="input is-rounded"
      onChange={[Function]}
      placeholder="Gif"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
